### PR TITLE
Add C wrapper of starknet_crypto for pedersen primitive 

### LINKED ISSUE #22: feat: implement the pedersen hash primitive
### Description
Use the [reference](https://docs.starkware.co/starkex/crypto/pedersen-hash-function.html) and implementation examples to implement the missing primitive. 

### Acceptance Criteria
Implement the logic like [here](https://github.com/starkware-libs/crypto-cpp/blob/78e3ed8dc7a0901fe6d62f4e99becc6e7936adfd/src/starkware/crypto/ffi/js/crypto.js#L37) and add tests to ensure it works correctly.

### PR DESCRIPTION
**Pull request by edisontim**
Closes #22

### CODE REVIEW

**Comment by edisontim** [comment]
Please do not merge this PR yet, running ```zig test src/tests.zig``` does not compile yet

**Comment by edisontim** [comment]
I propose adding a Makefile to the project in this PR: 
As with the cairo-vm-go, we will need to run specific commands for the coverage, fibonaci demos, etc. 
In my case I had a need for a Makefile to build the starknet_crypto library from source

**Comment by edisontim** [comment]
It's also closing this issue: #254, I've added a filter flag directly in the build.zig

**Comment by edisontim** [comment]
Basing this off of cairo-vm-go repo, they use the starknet_crypto Rust library with a C wrapper. I've done the same here and am linking the archive file in the build.zig and adding the starknet_crypto.h to the include paths so I could use the C functions in Zig

File: src/math/crypto/pedersen_hash.zig [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+const Felt252 = @import("../fields/starknet.zig").Felt252;
+const expectEqual = std.testing.expectEqual;
+
+const starknet_crypto = @cImport(@cInclude("starknet_crypto.h"));
+
+pub fn pedersenHash(a: Felt252, b: Felt252) Felt252 {
+    // convert Felt252 to big endian byte array
+    var a_bytes = a.toBytesBe();
+    var b_bytes = b.toBytesBe();
+    var res = [_]u8{0} ** 32;
```
Comment (tcoratger) [inline src/math/crypto/pedersen_hash.zig:11]: 32 is used multiple times over the PR can this be declared as a global constant somewhere?

**Review by lana-shanghai — APPROVED** [review]
Address Thomas's comment otherwise LGTM

**Review by tcoratger — APPROVED** [review]

**Pull request merged**